### PR TITLE
rewrite *write function

### PR DESCRIPTION
**Pull request by xiaobiaozhao**
1. Resolve compile-time warning
2. Improves code robustness

### CODE REVIEW

**Review by git-hulk — DISMISSED** [review]

File: src/common/util.cc [hunk 1]
```diff
@@ -672,4 +672,28 @@ int aeWait(int fd, int mask, uint64_t timeout) {
     return retval;
   }
 }
+
+Status Write(int fd, const std::string &data) {
```
Comment (PragmaTwice) [inline src/common/util.cc:676]: I wonder if there is any difference between `SockSend` and this new function.

Comment (xiaobiaozhao) [reply src/common/util.cc:676]: I'm going to create a pr that fixes the socksend function and replaces write with send. Becase send has more flags than write.

File: src/common/util.cc [hunk 2]
```diff
@@ -672,4 +672,28 @@ int aeWait(int fd, int mask, uint64_t timeout) {
...
+    ssize_t nwritten = write(fd, data.c_str() + n, data.size() - n);
+    if (nwritten == -1) {
+      return Status(Status::NotOK, strerror(errno));
+    }
+    n += nwritten;
+  }
+  return Status::OK();
+}
+
+Status Pwrite(int fd, const std::string &data, std::istream::off_type offset) {
```
Comment (PragmaTwice) [inline src/common/util.cc]:
```suggestion
Status Pwrite(int fd, const std::string &data, off_t offset) {
```
This function is not related to `std::istream`

File: src/common/util.cc (hunk 1 above)
Comment (PragmaTwice) [reply src/common/util.cc:676]: Do we really use the flag provided by `send`?

Currently `SockSend` has a totally same function body as this function. It is weird.

File: src/common/util.h [hunk 3]
```diff
@@ -75,5 +75,7 @@ auto GetTimeStamp() {
 }
 inline uint64_t GetTimeStampMS() { return GetTimeStamp<std::chrono::milliseconds>(); }
 inline uint64_t GetTimeStampUS() { return GetTimeStamp<std::chrono::microseconds>(); }
-
+// file util
+Status Write(int fd, const std::string &data);
+Status Pwrite(int fd, const std::string &data, std::istream::off_type offset);
```
Comment (PragmaTwice) [inline src/common/util.h]: Same as above.

**Review by caipengbo — DISMISSED** [review]

File: src/common/util.cc (hunk 2 above)
Comment (xiaobiaozhao) [reply src/common/util.cc]: done

**Review by git-hulk — APPROVED** [review]

**Review by caipengbo — APPROVED** [review]

**Review by PragmaTwice — APPROVED** [review]
Thanks. Generally looks good to me.
But as I mentioned above, currently `SockSend` has a totally same function body as the newly added function `Write`. So I will do some code clean if no further PR is proposed.

**Comment by git-hulk** [comment]
Thanks all, merging..
